src/Chapter1.hs

```diff
@@ -554,7 +554,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x > y && y > z = y
+    | z > y && y > x = y
+    | y > x && x > z = x
+    | z > x && x > z = x
+    | x > z && z > y = z
+    | y > z && z > y = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂

Here you covered all cases for sure. So if pattern matching didn't work on previous 4 cases, then you can say even without condition checking that the answer would be `z`:

```suggestion
    | otherwise = z
```